flexget/plugins/clients/aria2.py

```diff
@@ -105,22 +114,60 @@ def on_task_output(self, task, config):
                 logger.opt(exception=True).debug('Exception type {}', type(e))
                 raise
 
-    def add_entry(self, aria2, entry, config):
+    def add_entry(self, aria2, entry, config, task):
         """
         Add entry to Aria2
         """
         options = config['options']
         try:
-            options['dir'] = os.path.expanduser(entry.render(config['path']).rstrip('/'))
+            path = entry.get('path', config.get('path', None))
+            options['dir'] = os.path.expanduser(entry.render(path).rstrip('/'))
         except RenderError as e:
             entry.fail('failed to render \'path\': %s' % e)
             return
-        if 'filename' in config:
+
+        filename = entry.get('content_filename', config.get('filename', None))
+        add_extension = entry.get('content_extension', config.get('add_extension', False))
+
+        if filename:
+            if add_extension:
+                if isinstance(add_extension, bool):
+                    logger.debug('Getting filename from `{}`', entry['url'])
+                    content_disposition = None
+                    try:
+                        response = task.requests.get(entry['url'], headers=None, stream=True)
```

Comment:
If a request is opened with the stream parameter, it has to be closed manually. See https://docs.python-requests.org/en/latest/user/advanced/#body-content-workflow